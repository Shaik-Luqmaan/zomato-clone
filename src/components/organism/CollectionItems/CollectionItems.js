import React from 'react'
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import CollectionCard from '../../molecules/CollectionCard/CollectionCard';
import col1 from '../../pictures/sweet.jpg';
import col2 from '../../pictures/most.jpg';
import col3 from '../../pictures/trending.jpg';
import col4 from '../../pictures/collection_1.jpg';

const useStyles = makeStyles({
    root: {
      padding: "30px 40px 40px 20px",
    },
  });

function CollectionItems(props) {
const classes = useStyles();
// const {CollectionList} = props;

return (
    <Grid container className={classes.root} spacing={4}>
      {CollectionList.map((value, index) => {
        return (
          <Grid item md={3} sm={6} xs={12} key={index}>
              <CollectionCard {...value} />
          </Grid>
        );
      })}
    </Grid>
  );   
}

const CollectionList= [
  {
      text: "Sweet Tooth 2020",
      placesCount : '9 Places',
      img: col1,
  },
  {
      
      text: "Most Searched 2020",
      placesCount : '28 Places',
      img: col2,
      
  },
  {
      
      text: "Best Openings of 2020",
      placesCount : '25 Places',
      img: col3,
      
  },
  {
      
      text: "Trending this Week",
      placesCount : '28 Places',
      img: col4,
      
  },
]
export default CollectionItems
