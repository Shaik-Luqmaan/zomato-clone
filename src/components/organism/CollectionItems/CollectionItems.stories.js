import React from "react";
import CollectionItems from "./CollectionItems";


export default {
    title: "organisms/CollectionItems",
    component: CollectionItems,
}


const Template = (args)=>(
    <CollectionItems {...args} />
);


export const Primary = Template.bind({});
Primary.args = {
  
}