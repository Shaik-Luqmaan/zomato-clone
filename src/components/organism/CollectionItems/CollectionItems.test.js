import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom"
import renderer from 'react-test-renderer';
import CollectionItems from './CollectionItems';

afterEach(cleanup);

it("renders component without crashing",()=>{
    const div = document.createElement('div');
    ReactDOM.render(<CollectionItems/>, div)
})

it("renders CollectionItems snapshot correctly", () => {
    const tree = renderer.create(<CollectionItems />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  