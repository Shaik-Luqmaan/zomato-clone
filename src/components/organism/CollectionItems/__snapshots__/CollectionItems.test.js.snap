// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CollectionItems snapshot correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-4"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <div
      className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
      data-testid="main"
      style={
        Object {
          "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(sweet.jpg)",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-content-5 MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          data-testid="text"
        >
          Sweet Tooth 2020
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
          data-testid="place"
        >
          9 Places
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-arrow-3 MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 17l5-5-5-5v10z"
            />
          </svg>
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <div
      className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
      data-testid="main"
      style={
        Object {
          "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(most.jpg)",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-content-5 MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          data-testid="text"
        >
          Most Searched 2020
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
          data-testid="place"
        >
          28 Places
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-arrow-3 MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 17l5-5-5-5v10z"
            />
          </svg>
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <div
      className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
      data-testid="main"
      style={
        Object {
          "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(trending.jpg)",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-content-5 MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          data-testid="text"
        >
          Best Openings of 2020
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
          data-testid="place"
        >
          25 Places
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-arrow-3 MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 17l5-5-5-5v10z"
            />
          </svg>
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <div
      className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
      data-testid="main"
      style={
        Object {
          "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection_1.jpg)",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-content-5 MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          data-testid="text"
        >
          Trending this Week
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
          data-testid="place"
        >
          28 Places
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-arrow-3 MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 17l5-5-5-5v10z"
            />
          </svg>
        </h6>
      </div>
    </div>
  </div>
</div>
`;
